edx_lint/pylint/unicode_check.py

```diff
@@ -58,7 +58,22 @@ def process_tokens(self, tokens):
             # If the whole module is __future__'d, then it's all fine.
             return
 
-        for tok_type, tok_text, start, _, _ in tokens:
+        previous_line = ""
+        current_line = ""
+        first_non_comment_line = 0
+        for index, (tok_type, tok_text, start, _, _) in enumerate(tokens):
+            # Use new line characters to build previous line
+            if first_non_comment_line == 0 and tok_type != 53: # 53 is token.COMMENT
+                first_non_comment_line = index
+
+            if tok_type == token.NEWLINE or tok_type == 54:  # 54 is token.NL
+                previous_line = current_line
+                current_line = ""
+            else:
+                current_line += tok_text
+                if tok_text in ['def', 'class']:
+                    current_line += " "
```

Comment:
We only need `current_line` for checking if a previous line is a definition.  It could be simpler to just always add a space between tokens, and then update the regex to match the result.  Or: don't keep the previous line text at all, just remember if there was a keyword "def" or "class" token in the previous line.

Reply:
i was just adding the space after those tokens for my regex so it wouldn't get confused by functions or variables that contain those strings as substring.  
true, i can update the code to only check for those tokens

Reply:
updated the code